libc-test/test/style.rs

```diff
@@ -0,0 +1,357 @@
+//! Simple script to verify the coding style of this library
+//!
+//! ## How to run
+//!
+//! The first argument to this script is the directory to run on, so running
+//! this script should be as simple as:
+//!
+//! ```notrust
+//! cargo test --test style -- ../src
+//! ```
+//!
+//! ## Guidelines
+//!
+//! The current style is:
+//!
+//! * Specific module layout:
+//!     1. use directives
+//!     2. typedefs
+//!     3. structs
+//!     4. constants
+//!     5. f! { ... } functions
+//!     6. extern functions
+//!     7. modules + pub use
+//!
+//! Things not verified:
+//!
+//! * alignment
+//! * leading colons on paths
+
+use std::io::prelude::*;
+use std::ops::Deref;
+use std::path::Path;
+use std::{env, fs};
+
+use syn::parse::{Parse, ParseStream};
+use syn::spanned::Spanned;
+use syn::visit::{self, Visit};
+use syn::Token;
+
+macro_rules! t {
+    ($e:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed with {}", stringify!($e), e),
+        }
+    };
+}
+
+#[test]
+fn check_style() {
+    if env::var("LIBC_CI").is_ok() {
+        // we already run this in the style.sh script
+        return;
+    }
```

Comment:
CI is failing since this env var doesn't seem to be set. It seems like it might be specific to android, running it locally I get:

```
WARNING: linker: /data/local/tmp/style-d96f44504d869edb: unsupported flags DT_FLAGS_1=0x8000001
```